Update bug report responsibilities for Product Designers (#44278)
fleetdm/fleet: handbook/company/product-groups.md

At this state, the Head of Product Design is responsible for going through the inbox and adding the correct product group label (e.g. `#g-mdm`, `#g-orchestration`, `#g-software`, `#g-security-compliance`). 

> Then, it's the product group Product Designer's responsibility to decide if it's a bug, make sure reproduction steps are documented, and that the "🛠️ To fix" section is filled if the expected behavior is unclear. If the expected behavior is clear, the Tech Lead should provide the specified fix in the "🛠️ To fix" section.

Reviewer: ```suggestion
Then, it's the product group Product Designer's responsibility to decide if it's a bug, specify the expected behavior, and make sure reproduction steps are documented. If the expected behavior is unclear, ask the product group's Tech Lead for help.
```

@marko-lisica thanks! I made the instructions more generic so we don't have to worry about the handbook getting stale when we inevitably decide to rename "🛠️ To fix" section again (see below).

Re the bug's title, I hear you. The goal is to make it obvious to the rest of the business/community what the expected behavior is. Instead of updating the bug title (extra paper work), I think we retitle the "🛠️ To fix" section to "🛠️ Expected behavior". I added that to this PR.

What do you think?

cc @rachaelshaw @melpike
